pkg/apis/core/validation/validation.go

```diff
@@ -3312,6 +3312,19 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod) field.ErrorList {
 	mungedPod.Spec.Tolerations = oldPod.Spec.Tolerations
 	allErrs = append(allErrs, validateOnlyAddedTolerations(newPod.Spec.Tolerations, oldPod.Spec.Tolerations, specPath.Child("tolerations"))...)
 
+	// Backward compatibility with clients before v1.7.8t
+	for idx, vol := range oldPod.Spec.Volumes {
+		// If EmptyDir.Medium is not set to default value "", don't do convertion
+		if vol.EmptyDir.Medium != "" {
+			if mungedPod.Spec.Volumes[idx].EmptyDir.SizeLimit != nil {
+				// If EmptyDir.SizeLimit is set to "0", then take SizeLimit as a nil value
+				if mungedPod.Spec.Volumes[idx].EmptyDir.SizeLimit.String() == "0" && vol.EmptyDir.SizeLimit == nil {
+					mungedPod.Spec.Volumes[idx].EmptyDir.SizeLimit = nil
```

Comment:
This line should already be dropped by this line prior to validation:

https://github.com/kubernetes/kubernetes/blob/c933067cd3cfa81349446b929816f732aa145b90/pkg/api/pod/util.go#L242

Reply:
It seems that this `DropDisabledAlphaFields` function is after 1.8.0 release.
But [v1.7.8~v1.8.0) didn't have this validation, can I submit this PR onto `release-1.7 branch`?

Reply:
It would be better to add the DropDisabledAlphaFields part to the 1.7.x branch than this

Reply:
Ok, I will update this PR dat way.

Reply:
Hi, Could you review my PR again plz? Thx